generator/EnumGen.cs

```diff
@@ -25,49 +25,103 @@
 
 namespace GtkSharp.Generation {
 
-	public class EnumGen : GenBase {
+	// <enumeration name="BookmarkFileError" c:type="GBookmarkFileError" glib:error-domain="g-bookmark-file-error-quark">
+	//   <doc xml:space="preserve">Error codes returned by bookmark file parsing.</doc>
+	//   <member name = "invalid_uri" value="0" c:identifier="G_BOOKMARK_FILE_ERROR_INVALID_URI">
+	//     <doc xml:space="preserve">URI was ill-formed</doc>
+	//   </member>
+	public class EnumGen : GenBase
+	{
+		// TODO:
+		// See marshalling without casting
+
+		class EnumMember
+		{
+			public string Name { get; private set; }
+			public string Value { get; private set; }
+			bool isBitField;
+
+			public EnumMember (XmlElement elem, bool isBitField)
+			{
+				this.isBitField = isBitField;
+
+				Name = elem.GetAttribute (Constants.Name);
+				Value = elem.GetAttribute (Constants.Value);
+			}
+
+			public bool Validate ()
+			{
+				if (IsUnsigned) {
+					Value = Value.TrimEnd ('u', 'U');
+				}
+				if (isBitField) {
+					int value = int.Parse (Value);
+					Value = string.Format ("0x{0:X}", value);
+				}
+				return true;
+			}
 
-		string enum_type = String.Empty;
-		List<string> members = new List<string> ();
+			public bool IsUnsigned {

```

Comment:
Does this really happen in gir? Afaik enumerations are always int in gir and are not suffixed.

Reply:
I kept the code for backwards compat. I can definitely remove this, but just wanted to make sure this stayed in until I saw it's actually never unsigned.

Is there any spec that says it's never unsigned?

Reply:
Unfortunately, I don't think there is a spec but looking at the girscanner there is no attribute that could make a enum unsigned. Also I'm quite sure that I never saw such things happening in the .gir files I was converting.